src/components/movieDetailsComponent/MovieDetailsComponent.tsx

```diff
@@ -0,0 +1,172 @@
+import {useEffect, useState} from 'react';
+import {Image, ScrollView, Text, View} from 'react-native';
+import {RouteProp, useRoute} from '@react-navigation/native';
+
+import Icon from 'react-native-vector-icons/MaterialIcons';
+
+import {imageUrl} from '@helpers/helper';
+import {fetchData} from '@network/apiMethods';
+import HorizontalMovieScroll from '@components/horizontalMovieScroll/HorizontalMovieScroll';
+
+import {styles} from './styles';
+
+const MovieDetailsComponent = () => {
+  const route = useRoute<RouteProp<MovieScreenParamList, 'MOVIE_DETAILS'>>();
+  const movieId = route?.params?.movieId;
+
+  const [movieDetails, setMovieDetails] = useState<MovieCardData>({
+    id: movieId,
+    title: '',
+    poster_path: '',
+    backdrop_path: '',
+    overview: '',
+    vote_average: 0,
+    vote_count: 0,
+    release_date: '',
+    genre_ids: [],
+    adult: false,
+  });
+
+  const [recommendation, setRecommendation] = useState([]);
+  const [genreArr, setGenreArr] = useState([]);
+
+  const getData = async () => {
+    const externalId = await fetchData(`/movie/${movieId}/external_ids`);
+
+    const movie = await fetchData(`/find/${externalId?.imdb_id}`, {
+      params: {external_source: 'imdb_id'},
+    });
+    const movieData = movie?.movie_results[0];
+
+    const genre = await fetchData('/genre/movie/list');
+    setGenreArr(genre?.genres);
+
+    const reccomendMovies = await fetchData(
+      `/movie/${movieId}/recommendations`,
+      {params: {page: '1'}},
+    );
+    setRecommendation(
+      reccomendMovies?.results?.map(
+        ({
+          id,
+          title,
+          poster_path,
+          overview,
+          vote_average,
+          vote_count,
+          release_date,
+        }: MovieCardData) => ({
+          id,
```

Comment:
what's the use of map here?

Reply:
To transform the results array from the reccomendMovies object into a new array of objects. 
Each object in results array have many more key-value pairs but I need only selected key-value pairs in my object.